src/compas_rhino/artists/robotmodelartist.py

```diff
@@ -109,28 +109,37 @@ def _exit_layer(self):
         self.redraw()
 
     def draw_collision(self):
+        """Draws all collision geometry of the robot model. Return list of newly created Rhino object's GUID."""
         collisions = super(RobotModelArtist, self).draw_collision()
         collisions = list(collisions)
 
         self._enter_layer()
 
+        new_guids = []
         for mesh in collisions:
-            self._add_mesh_to_doc(mesh)
+            guids = self._add_mesh_to_doc(mesh)
+            new_guids.extend(guids)
 
         self._exit_layer()
+        return new_guids
 
     def draw_visual(self):
+        """Draws all visual geometry of the robot model. Return list of newly created Rhino object's GUID."""
         visuals = super(RobotModelArtist, self).draw_visual()
         visuals = list(visuals)
 
         self._enter_layer()
 
+        new_guids = []
         for mesh in visuals:
-            self._add_mesh_to_doc(mesh)
+            guids = self._add_mesh_to_doc(mesh)
+            new_guids.extend(guids)
 
         self._exit_layer()
+        return new_guids
 
     def draw(self):
+        """Same as draw_visual."""
         self.draw_visual()
```

Comment:
I think to be the same as `draw_visual` it would need to be `return self.draw_visual()`.